Rename Wikiteam3 > MediaWiki Scraper

robkam opened the pull request:
Done in a hurry probably needs polishing. 
The parts about `prepare-for-publication` branch in readme.md needs some revising?
In readme.md the last paragraph doesn't yet include @NyaMisty @yzqzss @robkam
Please check pyproject.toml is correct

There is a bug (Git Bash, Windows 10) that needs fixing:
`$ pip install --force-reinstall dist/*.whl`
snipped
`ERROR: For req: mediawiki-scraper==3.0.0. Invalid script entry point: <ExportEntry dumpgenerator = mediawiki:None []> - A callable suffix is required. Cf https://packaging.python.org/specifications/entry-points/#use-for-scripts for more information.`

elsiehupp commented:
Just on first glance at the files changed, in the README, where it says "`MediaWiki Scraper`" it can instead say "MediaWiki Scraper" (i.e. without the tick signs).

Also I changed the URL, so you can safely make it `https://github.com/mediawiki-client-tools/mediawiki-scraper`.

It might useful to change the name of `dumpgenerator`, but that can probably go in a separate Pull Request?

robkam commented:
I've removed the tick marks. Yes to rename dumpgenerator, it'll be easier to get this PR out of the way first. I've changed the link in the about but that might not get across.

elsiehupp commented on mediawiki-scraper/launcher.py:
```diff
@@ -25,8 +25,8 @@
 
-from wikiteam3.dumpgenerator.config import Config
-from wikiteam3.utils import domain2prefix
+from mediawiki-scraper.dumpgenerator.config import Config
```
Can you test to make sure the hyphenated version of the name is correct? IIRC package names have to use underscores rather than hyphens. (Basically just run a sample command from the command line if you don't have an interactive development environment set up.)

robkam replied:
I'll check but it will take a little while, I'm getting out of my depth here. Won't it mess things up if file names are hyphens and package names are underscores?

NyaMisty commented:
Don't merge, do not use slash

NyaMisty commented:
Why don't we simply call wikidump lol

robkam commented:
[PEP 8](https://peps.python.org/pep-0008/#package-and-module-names): Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.

NyaMisty commented:
I suggest we postpone the renaming? We can still discuss for  a while as we keep refactor & improve this project to make it ready for publish. 
We can try to contact if ArchiveTeam wants us lol

robkam commented:
I've replaced all `from mediawiki-scraper` to `from mediawikiscraper`

robkam commented:
It now gets past `$ pip install --force-reinstall dist/*.whl` but there's still a (probably basic) mistake somewhere.
```
$ dumpgenerator  --failfast --xml --xmlrevisions --images http://group2.mediawiki.demo.save-web.org/mediawiki-1.35.9/api.php
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Python\Scripts\dumpgenerator.exe\__main__.py", line 4, in <module>
ModuleNotFoundError: No module named 'mwscraper.dumpgenerator'
```

robkam commented:
Now works fine with `dumpgenerator  --failfast --xml --xmlrevisions --images http://group2.mediawiki.demo.save-web.org/mediawiki-1.35.9/api.php`

elsiehupp commented:
> I suggest we postpone the renaming? We can still discuss for a while as we keep refactor & improve this project to make it ready for publish. We can try to contact if ArchiveTeam wants us lol

The reason for the renaming is, specifically, to be able to deploy this package on PyPI, and to be able to do so without stepping on anyone's toes. I'm happy to rename the GitHub org and the repository yet again if there's some consensus around another name.

If you—@NyaMisty—want to reach out to the maintainer over at WikiTeam, Federico Leva (@nemobis), you can try the methods recommended on [his website's contact page](https://federicoleva.eu/en/contact/). I personally had success reaching him a while back via the email listed there. Basically you can ask him to weigh in on the question of what PyPI name(s) he thinks we should (or should not) use, in addition to asking about how he views the upstream repository's relationship to this one.

Again, the renaming is just a means for facilitating publication on PyPI, and publication on PyPI to me seems like an important milestone for us to pursue in the near-to-medium-term future.

Regarding merging this particular commit, I've personally taken enough of a backseat role at this point that I will wait for someone other than me to approve the merge. I don't feel comfortable (by any means!) telling the collaborators here what to do, and I see my role in this particular situation as primarily just doing the administrative IT tasks necessary to help move things along.

elsiehupp commented:
> [PEP 8](https://peps.python.org/pep-0008/#package-and-module-names): Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.

I'm not quite clear on the difference between "module names" and "package names"... my understanding is that `mediawiki_scraper` is legible, and it's specific enough to avoid any name collisions. In general it's easier to be pithy with a clever brand name (like [Pillow](https://pypi.org/project/Pillow/)), but I'm personally terrible at coming up with those, much less being confident in doing so... 😬

Again, if consensus emerges around another name, I'd be happy to do whatever IT admin tasks would be necessary to facilitate that.

robkam commented:
> Regarding merging this particular commit, I've personally taken enough of a backseat role at this point that I will wait for someone other than me to approve the merge.
#86 anyone? Please let's keep making progress on the port to Python 3.

NyaMisty commented:
Well. I still have something to say:
1. We still should avoid the dash or underscore:
  - @elsiehupp I know there're many packages on PyPI has dash in their package name. But AFAIK there EVEN MORE packages avoiding dashes in their package name. Among all package we use as dependency, only 2 in 12 are using dash or underscore. I thinks that already tells us what should do LOL.
  - Because you need underscore in package name and dash in PyPI&Project name, there will be confusion everywhere because the difference.
2. We shouldn't use too long name
  - `mediawiki_scraper.dumpgenerator` or `mediawikiscraper.dumpgenerator` is toooooooo long. It's very inconvenient for typing. You'll need to write these names in `from XXX import XXX` statements, `pip install XXX` commands, `python -m XXXX` and even more. We need a shorter name.
  - Personally I still want to keep the `dumpgenerator`, so maybe we can shorten the previous part ;)

3. We are stil far from stable and releasing.
  - There are still various glitches inside this project and we are still try our best to make --xmlrevision stable and working. I think we shouldn't post this tool onto PyPI before we successfully dumping a fairly big wiki site.

robkam commented:
I've used `mwscraper` instead of `mediawiki-scraper ` except in the name and in [pyproject.toml](https://github.com/mediawiki-client-tools/mediawiki-scraper/blob/python3/pyproject.toml)

elsiehupp commented:
> * `mediawiki_scraper.dumpgenerator` or `mediawikiscraper.dumpgenerator` is toooooooo long.

My thought was that `mediawiki_scraper` or similar would be the only package in this particular PyPI listing, so it would just be that (without the `.dumpgenerator` on the end).

That said, what about `wikiteam-scraper` on PyPI (and GitHub) and just `scraper` for the package name? (I mentioned this on the Issue, as well).

Regarding publishing on PyPI: I don't think this needs to be any more stable than the upstream project is (is or is not), and we could do a more moderate version-number bump (rather than resetting at 1.0.0).

PyPI does have [classifiers](https://pypi.org/classifiers/) for various stages of preproduction, so one of those could be added as appropriate.

As far as versioning goes, the two main reasons I had thought we could do a version-number reset to `1.0.0` are:

1. The Python 3 port _does_ break backwards compatibility (by virtue of being Python 3), which, under semantic versioning, requires a major version number bump, and
2. Pushing past the "1.0 barrier" would give us more flexibility to follow semantic-versioning guidelines, rather than artificially limiting ourselves to minor and patch number bumps in order to stay below `1.0`.
